sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -1015,6 +1020,10 @@ object DateTimeUtils {
     guess
   }
 
+  def convertTz(ts: SQLTimestamp, fromZone: String, toZone: String): SQLTimestamp = {
+    convertTz(ts, getTimeZone(fromZone), getTimeZone(toZone))
```

Comment:
performance is going to suck here

Reply:
I guess caching the value as done by the `FromUTCTimestamp` expression is the right way to go?